pkg/cache/v3/linear.go

```
@@ -298,10 +298,10 @@ func (cache *LinearCache) GetResources() map[string]types.Resource {
 	return resources
 }
 
-func (cache *LinearCache) CreateWatch(request *Request, _ stream.StreamState, value chan Response) func() {
+func (cache *LinearCache) CreateWatch(request *Request, _ SubscriptionState, value chan Response) (func(), error) {
 	if request.GetTypeUrl() != cache.typeURL {
 		value <- nil
-		return nil
+		return nil, fmt.Errorf("request type %s does not match cache type %s", request.TypeUrl, cache.typeURL)
```

Are there practical cases where this is important to catch this in a production system?

My understanding is that this would be the result of a muxer that is not configured properly, IMO it's not worth increasing the size of the interface for all the valid cases. I'll leave it up to you though.

---

We had issues on slack (and I think GitHub) due to such kind of silent errors.

---

OK, I'll let you discuss this with other maintainers.